source/contracts/reporting/Auction.sol

```diff
@@ -0,0 +1,239 @@
+    }
+
+    function getRepPriceInAttoEth() public view returns (uint256) {
+        // Use the manually provided REP price if the Auction system is not used for setting fees yet
+        if (!controller.useAuction()) {
+            return manualRepPriceInAttoEth;
+        }
+
+        // If this auction is over and it is a recorded auction use the price it found
+        if (getAuctionIndexForCurrentTime() != currentAuctionIndex && currentAuctionType == AuctionType.RECORDED) {
+            return currentRepPrice;
+        }
+
+        return repPrice;
+    }
+
+    function getAuctionIndexForCurrentTime() public view returns (uint256) {
+        return controller.getTimestamp().sub(initializationTime) / Reporting.getAuctionDuration();
+    }
+
+    function isActive() public view returns (bool) {
+        AuctionType _auctionType = getAuctionType();
+        return _auctionType == AuctionType.UNRECORDED || _auctionType == AuctionType.RECORDED;
+    }
+
+    function getAuctionType() public view returns (AuctionType) {
+        uint256 _auctionDay = getAuctionIndexForCurrentTime() % 7;
+        return AuctionType(_auctionDay);
+    }
+
+    function getAuctionStartTime() public view returns (uint256) {
+        uint256 _auctionIndex = getAuctionIndexForCurrentTime();
+        uint256 _auctionDay = _auctionIndex % 7;
+        uint256 _weekStart = initializationTime.add(_auctionIndex).sub(_auctionDay);
+        uint256 _addedTime = _auctionDay > uint256(AuctionType.UNRECORDED) ? uint256(AuctionType.RECORDED) : uint256(AuctionType.UNRECORDED);
+        return _weekStart.add(_addedTime.mul(Reporting.getAuctionDuration()));
+    }
+
+    function getAuctionEndTime() public view returns (uint256) {
+        uint256 _auctionIndex = getAuctionIndexForCurrentTime();
+        uint256 _auctionDay = _auctionIndex % 7;
+        uint256 _weekStart = initializationTime.add(_auctionIndex).sub(_auctionDay);
+        uint256 _addedTime = _auctionDay > uint256(AuctionType.UNRECORDED) ? uint256(AuctionType.RECORDED) : uint256(AuctionType.UNRECORDED);
+        _addedTime += 1;
+        return _weekStart.add(_addedTime.mul(Reporting.getAuctionDuration()));
+    }
+
+    function getUniverse() public view afterInitialized returns (IUniverse) {
+        return universe;
+    }
+
+    function getReputationToken() public view afterInitialized returns (IReputationToken) {
+        return reputationToken;
+    }
+
+    function setRepPriceInAttoEth(uint256 _repPriceInAttoEth) external afterInitialized onlyAuthorizedPriceFeeder returns (bool) {
+        manualRepPriceInAttoEth = _repPriceInAttoEth;
+        return true;
+    }
+
+    function addToAuthorizedPriceFeeders(address _priceFeeder) public afterInitialized onlyKeyHolder returns (bool) {
+        authorizedPriceFeeders[_priceFeeder] = true;
+        return true;
+    }
+
+    function removeFromAuthorizedPriceFeeders(address _priceFeeder) public afterInitialized onlyKeyHolder returns (bool) {
```

Comment:
Naming: This function name shouldn't be pluralized, it only removes a single `priceFeeder`.  Same with previous function.

Reply:
I think the name makes sense since it isn't `removeAuthorizedPriceFeed` it's remove from a larger set?

Reply:
The name is is remove _from_ authorized price feeders. Could rename it to just `removeAuthorizedPriceFeeder`

Reply:
Ah, I didn't catch that.  I reasonably strongly thing this should be renamed as @nuevoalex suggested since right now the function name is referencing an internal (private) datastructure as its name, rather than the publicly visible concept of price feeders.